### PR TITLE
[HttpClient] Throw JsonException instead of TransportException on empty response in Response::toArray()

### LINKED ISSUE #37064: [HttpClient] Response::toArray() throws TransportException instead of JsonException when response body is empty
**Symfony version(s) affected**: 4.3.0 - 5.1.0

**Description**  
When calling `toArray()` on a response with an empty body, a `TransportException` is thrown. A `TransportException` should be thrown on network errors. However, responses with empty body might be completely valid. In that case, `toArray()` should throw `JsonException` (or other exception that implements `DecodingExceptionInterface`) should be thrown, because the reponse body does not contain valid JSON.

**How to reproduce**  
```
$httpClient = HttpClient::create();
$response = $httpClient->request('GET', 'https://httpbin.org/status/201');
$response->toArray(); // throws TransportException with message 'Response body is empty.'
```

**Possible Solution**  
Throw a `JsonException` instead of a `TransportException` when response body is empty.

### PR DESCRIPTION
**Pull request by jeroennoten**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37064
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/HttpClient/Tests/Response/MockResponseTest.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Test methods from Symfony\Component\HttpClient\Response\ResponseTrait.
+ */
+class MockResponseTest extends TestCase
+{
+    /**
+     * @dataProvider toArrayErrors
+     */
+    public function testToArrayError($content, $responseHeaders, $message)
```
Comment (jeroennoten) [inline src/Symfony/Component/HttpClient/Tests/Response/MockResponseTest.php:17]: This test method was copied from the master branch.

**Comment by javiereguiluz** [comment]
@jeroennoten thanks for this contribution! However, `4.3` branch is no longer maintained, so this should be submitted against `4.4` (see https://symfony.com/releases for the full details). Thanks.

**Comment by jeroennoten** [comment]
Hi @javiereguiluz, thanks for your feedback. I rebased to 4.4 and changed the target branch this pull request to 4.4.

**Review by nicolas-grekas — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jeroennoten.

**Pull request merged**